Modules/_ctypes/cfield.c

```diff
@@ -1087,6 +1091,34 @@ d_get(void *ptr, Py_ssize_t size)
     return PyFloat_FromDouble(val);
 }
 
+#ifdef __STDC_IEC_559_COMPLEX__
+#if defined(__clang__) && __has_builtin(__builtin_complex) && \
+    !defined(CMPLX)
+#  define CMPLX(x, y) __builtin_complex ((double) (x), (double) (y))
```

Comment:
For safety, I'd recommend checking the existence of `CMPLX` for `!defined(__clang__)` and raise and/or disable the complex support entirely or use supply the macro.

Reply:
Other compilers (than clang), that claims to implement C11 (which we require) *and* define ``__STDC_IEC_559_COMPLEX__`` don't have issue with CMPLX().  This is specific to glibc & clang combination: https://sourceware.org/bugzilla/show_bug.cgi?id=26287

Perhaps I should add a comment.

Unfortunately, we can't implement CMPLX() without ``__builtin_complex()``: almost all corner cases could be fixed, but nan in the imaginary component will break things.  (I doubt some elementary function from the complex.h could help to fix ``CMPLX(<finite>, NAN)`` case.)

Reply:
Ah, right `__builtin_complex` does more. Yes, a comment might be helpful.